### ISSUE TITLE
Unshaded JAR includes all shaded dependencies increasing build time by 50%

### ISSUE THREAD

**Issue by argium** [Azure/azure-cosmosdb-spark#377]
After upgrading to 3.0.0 my team noticed that build times increased by 50%.  I traced it to what seems to be lines starting with **cosmosdb_connector_shaded**:

```2020-05-20T15:53:24.1493898Z 15:53:24,136 [DEBUG] adding entry cosmosdb_connector_shaded/rx/internal/util/atomic/SpscLinkedArrayQueue.class```

We import the non-shaded version via:

```xml
                <cosmosdb.version>3.0.0</cosmosdb.version>

<snip>

        <dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>azure-cosmosdb-spark_${spark.version}_${scala.major.version}</artifactId>
            <version>${cosmosdb.version}</version>
        </dependency>
```

Looking at the file sizes on the [Central Repository](https://repo1.maven.org/maven2/com/microsoft/azure/azure-cosmosdb-spark_2.4.0_2.11/3.0.1/), they both look the same:

![Annotation 2020-05-20 130859](https://user-images.githubusercontent.com/15852038/82492525-32bbd380-9a9b-11ea-8e8c-0306a766c8e5.png)

Therefore I suspect the non-shaded version is including all of its dependencies when it should not.  I expect Maven to be able to resolve dependencies based on the rules in ours project's pom.xml file.

**Comment by argium** [comment]
I think it's a regression from 2.1.5.

![image](https://user-images.githubusercontent.com/15852038/82492948-f2108a00-9a9b-11ea-80d0-6c0fc28256bb.png)

**Comment by FabianMeiswinkel** [comment]
Fixed in V3.0.2
